model/infinispan/src/main/java/org/keycloak/keys/infinispan/InfinispanPublicKeyStorageProvider.java

```diff
@@ -124,22 +124,23 @@ public KeyWrapper getFirstPublicKey(String modelKey, String algorithm, PublicKey
 
     @Override
     public KeyWrapper getPublicKey(String modelKey, String kid, String algorithm, PublicKeyLoader loader) {
-        // Check if key is in cache
         PublicKeysEntry entry = keys.get(modelKey);
-        if (entry != null) {
+        int lastRequestTime = entry==null ? 0 : entry.getLastRequestTime();
+        int currentTime = Time.currentTime();
+        boolean isKeyCacheStale = currentTime > lastRequestTime + minTimeBetweenRequests;
```

Comment:
Minor: Maybe this boolean variable can be renamed to `isSendingRequestAllowed` ? As the cache entry is technically not state?